File: Week 6/pset6/readability-python/readability.py
import math
import string as strloop


def main():
    # Get the user input
    sentence = input("Sentence: ")
    grade = getGrade(sentence)

    # Different prints if it is greater than 15 or less than 1
    if grade < 1:
        print("Before Grade 1")
    elif grade > 15:
        print("Grade 16+")
    else:
        print(f'Grade {grade}')


def getGrade(string):
    sentences = 0
    letters = 0
    words = 1   # Last word will have no space so I need to start at 1

    # Run through each letter
    for letter in string:
        if letter == "!" or letter == "?" or letter == ".":
            sentences += 1
        elif letter in strloop.punctuation:
            continue
        elif letter in strloop.whitespace:
            words += 1
        else:
            letters += 1

    # Average letters and sentences per 100 words
    L = letters / words * 100
    S = sentences / words * 100

    # Coleman-Liau index formula
    return round(0.0588 * L - 0.296 * S - 15.8)


main()

File: Week 2/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int checkScore(string phrase);

int main(void)
{
    // Get text from the user
    string sentence = get_string("Input a sentence: ");

    // Find out the grade
    int grade = checkScore(sentence);

    // Print out the grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int checkScore(string phrase)
{
    /*
        index = 0.0588 * L - 0.296 * S - 15.8
        L is the average letters per 100 words
        S is average number of sentences per 100 words
    */


    // Next word is just a ' '
    // Iterate through every letter and when there's a space that's a new word
    // Every '.', '!', '?', is a new sentence

    int words = 1;
    int letters = 0;
    int sentences = 0;

    for (int i = 0; phrase[i] != '\0'; i++)
    {
        if (phrase[i] == ' ')
        {
            words++;
        }
        else if (phrase[i] == '!' || phrase[i] == '.' || phrase[i] == '?')
        {
            sentences++;
        }
        else if (tolower(phrase[i]) >= 'a' && tolower(phrase[i]) <= 'z')
        {
            letters++;
        }
    }

    // If I have 10 words I need to multiply by 10, I need 100 / number of words then multiply that by the final number of letters and sentences
    float avg = (float) 100.0 / words;
    float L = avg * letters;
    float S = avg * sentences;

    float score = (0.0588 * L) - (0.296 * S) - 15.8;

    // printf("%f\n", score);

    // Automatically round the number
    if ((int)(score * 10) % 10 >= 5)
    {
        score = (int) score + 1;
    }

    return (int) score;
}